patches/server/0969-New-event-verbose-logging-command.patch

```diff
@@ -0,0 +1,204 @@
++                MinecraftServer.LOGGER.warn("Error occurred while dumping event verbose", e);
++            }
++        }
++        return true;
++    }
++
++    /**
++     * Dumps all currently stored {@link Wrapper}'s to our {@link #currentFile}
++     * @throws IOException If something broke writing it
++     */
++    private void logWrappers() throws IOException {
++        JsonArray array = new JsonArray();
++        toDump.forEach(w -> {
++            @Nullable JsonObject obj = w.serialize();
++            if(obj != null) //will only be null if an event class only has final fields
++                array.add(obj);
++        });
++
++        StringWriter stringWriter = new StringWriter();
++        JsonWriter jsonWriter = new JsonWriter(stringWriter);
++        jsonWriter.setIndent(" ");
++        jsonWriter.setLenient(false);
++        Streams.write(array, jsonWriter);
++
++        try (PrintStream out = new PrintStream(Files.newOutputStream(currentFile), false, StandardCharsets.UTF_8)) {
++            out.print(stringWriter);
++        }
++
++    }
++
++    /**
++     * Called in {@link io.papermc.paper.plugin.manager.PaperPluginManagerImpl#callEvent(Event)} to save in {@link #toDump}
++     */
++    public static void tryLogEvent(Event ev) {
++        if(!isRunning)
++            return;
++        if(exclude.contains(ev.getClass()))
++            return;
++        toDump.add(new Wrapper(ev));
++    }
++
++    /**
++     * Wrapper class for QoL Serialization, just stores the event object and the time it ran to log eventually
++     */
++    private static final class Wrapper {
++        private final String timestamp;
++        private final Event event;
++
++        public Wrapper(Event event) {
++            this.timestamp = FORMATTER.format(LocalDateTime.now());
++            this.event = event;
++        }
++
++        @Nullable
++        /**
++         * Properly serialize our Wrapper into a JsonObject, so we can log it properly
++         * @implNote Is null if the event has only final fields
++         */
++        private JsonObject serialize() {
++            JsonObject parent = new JsonObject();
++            parent.addProperty("timestamp", timestamp);
++            JsonObject obj = new JsonObject();
++            obj.addProperty("event", event.getEventName());
++            try {
++                Field[] fields = event.getClass().getDeclaredFields();
++                //ignore update events / anything unchangeable except for ItemStacks, Collections and Maps
++                if(Stream.of(fields).allMatch(f -> Modifier.isFinal(f.getModifiers()) && !f.getType().equals(ItemStack.class) &&
++                    !Map.class.isAssignableFrom(f.getType()) && !Collection.class.isAssignableFrom(f.getType())))
++                    return null;
++                for (Field f : fields) {
++                    if(f.getName().equalsIgnoreCase("handlers"))
```

Comment:
This is a bad way to ignore the handler list fields. You should just ignore "static" fields